File: app/src/am_viennv/com/example/asian/database/UserDao.java
package com.example.asian.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.asian.beans.User;

import java.util.List;

@Dao
public interface UserDao {
    @Query("SELECT * FROM user")
    List<User> getAll();

    @Insert
    void insertUser(User user);

    @Query("SELECT * FROM user WHERE user_name LIKE '%' || :charToFind || '%'")
    List<User> findUserByName(String charToFind);

    @Query("DELETE FROM user")
    void deleteAll();

    @Delete
    void delete(User user);
}

File: Kotlin/Ex02/app/src/main/java/com/example/roomdb/model/UserDao.kt
package com.example.roomdb.model

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface UserDao {
    @Query("SELECT * FROM users")
    fun readAllUsers(): LiveData<List<User>>

    /* @Query("SELECT * FROM student_table WHERE uid IN (:userIds)")
     fun loadAllByIds(userIds: IntArray): List<Student>*/

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(user: User)

    @Delete
    suspend fun delete(user: User)

    @Query("DELETE FROM users")
    suspend fun deleteAll()
}